src/Model/Metadata/Helper/MDParser.php

```diff
@@ -52,6 +52,59 @@ public function parseAPIResponseEvent(array $response): Metadata
             );
         }
 
+        $catalogue = $this->catalogueFactory->event();
+        $metadata = $this->metadataFactory->event();
+
+        return $this->parseAPIResponseGeneric($fields, $metadata, $catalogue);
+    }
+
+    /**
+     * @param stdClass $data
+     * @return Metadata
+     * @throws \xoctException
+     */
+    public function getMetadataFromData($data) : Metadata
+    {
+        $fields = [];
+
+        foreach ($data as $key => $entry) {
+            switch ($key) {
+                case "presenter":
+                    $field_data = (object) array(
+                        "id" => "creator",
+                        "value" => $entry
+                    );
+                    break;
+                case "creator":
+                    $field_data = (object) array(
+                        "id" => "publisher",
+                        "value" => $entry
+                    );
+                    break;
+                case "is_part_of":
+                    $field_data = (object) array(
+                        "id" => "isPartOf",
+                        "value" => $entry
+                    );
+                    break;
+                case "start":
+                    $field_data = (object) array(
+                        "id" => "startDate",
+                        "value" => $entry
+                    );
+                    break;
+                default:
+                    $field_data = (object) array(
+                        "id" => $key,
+                        "value" => $entry
+                    );
+
+            }
+            if (!in_array($field_data, $fields)) {
+                array_push($fields, $field_data);
+            }
+
+        }
         $catalogue = $this->catalogueFactory->event();
         $metadata = $this->metadataFactory->event();
         return $this->parseAPIResponseGeneric($fields, $metadata, $catalogue);
```

Comment:
Those three lines are duplicate code. Please factor out in new protected method.